### ISSUE TITLE
Functions UEFI.read_EFI_variables_from_SPI and dump_EFI_variables_from_SPI are broken

### ISSUE THREAD

**Issue by fishilico** [chipsec/chipsec#1844]
Hello,
While trying to add type hints to many modules, I encountered an issue in https://github.com/chipsec/chipsec/blob/36ca6fa5a78433d26ca8adea622afce56cc15bf7/chipsec/hal/uefi.py#L352-L361

`self.cs.spi` does not exist. When trying to call this function with

```python
# ... (initialize cs using chipsec.chipset.cs())
uefi = chipsec.hal.uefi.UEFI(cs)
uefi.dump_EFI_variables_from_SPI()
```

An `AttributeError` exception is raised:

```text
AttributeError: 'Chipset' object has no attribute 'spi'
```

Moreover the call to `func_getefivariables` is never actually done, in `read_EFI_variables_from_SPI` and `read_EFI_variables_from_file`: https://github.com/chipsec/chipsec/blob/36ca6fa5a78433d26ca8adea622afce56cc15bf7/chipsec/hal/uefi.py#L366-L368

I think `(efi_var_store)` is missing:

```python
efi_vars = uefi_platform.EFI_VAR_DICT[self._FWType]['func_getefivariables'](efi_var_store)
```

In short, functions `dump_EFI_variables_from_SPI`, `read_EFI_variables_from_SPI` and `read_EFI_variables_from_file` do not work. As they are not currently used in chipsec, should them be removed?
